StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1117ParametersMustBeOnSameLineOrSeparateLines.cs

```diff
@@ -290,81 +218,44 @@ private static void HandleParameterListSyntax(SyntaxNodeAnalysisContext context,
             }
 
             SeparatedSyntaxList<ParameterSyntax> parameters = parameterList.Parameters;
-            if (parameters.Count > 2)
-            {
-                Analyze(context, parameters);
-            }
+            Analyze(context, parameters);
         }
 
         private static void HandleBracketedArgumentListSyntax(SyntaxNodeAnalysisContext context, BracketedArgumentListSyntax bracketedArgumentList)
         {
             SeparatedSyntaxList<ArgumentSyntax> arguments = bracketedArgumentList.Arguments;
-            if (arguments.Count > 2)
-            {
-                Analyze(context, arguments);
-            }
+            Analyze(context, arguments);
         }
 
-        private static void Analyze(SyntaxNodeAnalysisContext context, SeparatedSyntaxList<ParameterSyntax> parameters)
+        private static void Analyze(SyntaxNodeAnalysisContext context, IReadOnlyList<SyntaxNode> arguments)
```

Comment:
💡 This can be made generic to avoid boxing at the call sites.

```suggestion
        private static void Analyze<T>(SyntaxNodeAnalysisContext context, SeparatedSyntaxList<T> arguments)
            where T : SyntaxNode
```

Reply:
➡️ Fixed!

@pantosha This was a case where the original code was _unnecessarily_ complex to apply a performance optimization. Your initial change was a dramatic improvement to readability, and together we also made it just as fast 👍